### PR TITLE
Feat: clean up the canary-related resources while canary step's weight and matches were nil

### PR DESCRIPTION
**Pull request by yangsoon**
### Ⅰ. Describe what this PR does
this PR will clean up the canary-related resources while canary step's weight and matches are nil

### CODE REVIEW

File: pkg/trafficrouting/manager.go [hunk 1]
```diff
@@ -171,6 +175,26 @@ func (m *Manager) FinalisingTrafficRouting(c *util.RolloutContext, onlyRestoreSt
 	if trafficRouting.GracePeriodSeconds <= 0 {
```
Comment (zmberg) [inline pkg/trafficrouting/manager.go:175]:
```
trafficRouting := c.Rollout.Spec.Strategy.Canary.TrafficRoutings[0]
	cServiceName := fmt.Sprintf("%s-canary", trafficRouting.Service)
	// fetch canary service
	canaryService := &corev1.Service{}
	err := m.Get(context.TODO(), client.ObjectKey{Namespace: c.Rollout.Namespace, Name: cServiceName}, canaryService)
	if err != nil {
		if errors.IsNotFound(err) {
			return true, nil
		}
		klog.Errorf("rollout(%s/%s) get canary service(%s) failed: %s", c.Rollout.Namespace, c.Rollout.Name, cServiceName, err.Error())
		return false, err
	}
```

Comment (yangsoon) [reply pkg/trafficrouting/manager.go:175]: got it

File: pkg/webhook/rollout/validating/rollout_create_update_handler.go [hunk 2]
```diff
@@ -203,15 +203,23 @@ func validateRolloutSpecCanaryTraffic(traffic *appsv1alpha1.TrafficRouting, fldP
...
+	if len(routings) > 0 {
+		isTraffic = true
+		if routings[0].Ingress != nil {
+			ingressClassType = routings[0].Ingress.ClassType
+		}
+	}
 	for i := range steps {
 		s := &steps[i]
-		if isTraffic && s.Weight == nil && len(s.Matches) == 0 {
+		if isTraffic && s.Weight == nil && len(s.Matches) == 0 && ingressClassType != "aliyun-alb" {
```
Comment (furykerry) [inline pkg/webhook/rollout/validating/rollout_create_update_handler.go:222]: what's different about aliyun-alb, plz comment in the code. in general one should not hard code such provider type in the code

Comment (yangsoon) [reply pkg/webhook/rollout/validating/rollout_create_update_handler.go]: sorry, this code has removed

**Pull request merged**